packages/typespec-azure-resource-manager/test/rules/arm-delete-response-codes.test.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  BasicTestRunner,
+  LinterRuleTester,
+  createLinterRuleTester,
+} from "@typespec/compiler/testing";
+import { beforeEach, it } from "vitest";
+import { armDeleteResponseCodesRule } from "../../src/rules/arm-delete-response-codes.js";
+import { createAzureResourceManagerTestRunner } from "../test-host.js";
+
+let runner: BasicTestRunner;
+let tester: LinterRuleTester;
+
+beforeEach(async () => {
+  runner = await createAzureResourceManagerTestRunner();
+  tester = createLinterRuleTester(
+    runner,
+    armDeleteResponseCodesRule,
+    "@azure-tools/typespec-azure-resource-manager"
+  );
+});
+
+it("Emits a warning for synchronous delete operation that does not contain the appropriate response codes", async () => {
+  await tester
+    .expect(
+      `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Contoso;
+      
+      model Employee is ProxyResource<EmployeeProperties> {
+        @doc("Name of employee")
+        @pattern("^[a-zA-Z0-9-]{3,24}$")
+        @key("employeeName")
+        @path
+        @segment("employees")
+        name: string;
+      }
+      
+      model EmployeeProperties {}
+      
+      @armResourceOperations
+      interface Employees {
+        @armResourceDelete(Employee)
+        delete(...ApiVersionParameter): {
+          @statusCode _: 200;
+          result: boolean;
+        }
+      }
+    `
+    )
+    .toEmitDiagnostics({
+      code: "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes",
+      message:
+        "Synchronous delete operations must have 200, 204 and default responses. They must not have any other responses.",
+    });
+});
+
+it("Does not emit a warning for synchronous delete operation that contains the appropriate response codes", async () => {
+  await tester
+    .expect(
+      `
+    @armProviderNamespace
+    @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+    namespace Microsoft.Contoso;
+    
+    model Employee is ProxyResource<EmployeeProperties> {
+      @doc("Name of employee")
+      @pattern("^[a-zA-Z0-9-]{3,24}$")
+      @key("employeeName")
+      @path
+      @segment("employees")
+      name: string;
+    }
+    
+    model EmployeeProperties {}
+    
+    @armResourceOperations
+    interface Employees {
+      @armResourceDelete(Employee)
```

Comment:
We should add a test that validates ArmResourceDeleteSync does not throw the error, as regression protection